legacy/dockerregistry/inventory.go

```diff
@@ -1607,6 +1608,9 @@ func MkReadRepositoryCmdReal(
 			repoPath)
 	}
 
+	// Log the HTTP request to GCR.
+	reqcounter.Increment()
```

Comment:
Looking at this code again, we should instead just call reqcounter.Increment() once, inside the `ExecRequests()` function, inside the `for reqRes := range requestResults {` loop. This is both simpler (it's only 1 place we have to increment instead of 2 places), and also reuses the RequestResult channel which already acts as a global "funnel" where all requests, *after they've been processed*, comes in. Another reason we should do it there is that it will work for both real HTTP requests, and also fake ones we generate during unit tests (so it appears this was bound to come up anyway).

I should have taken a closer look the first time around, so this is my bad. But please do address it! :)

Reply:
Thank you! I'm sorry I didn't ask this myself. I was actually unsure where to put this counter in the first place. This is very helpful.